common/lib/xmodule/xmodule/modulestore/tests/django_utils.py

```diff
@@ -338,6 +338,7 @@ def create_user_for_course(self, course, user_type=CourseUserType.ENROLLED):
         Create a test user for a course.
         """
         if user_type is CourseUserType.ANONYMOUS:
+            self.client.logout()
```

Comment:
The user is already logged out if the user type is ANONYMOUS?

Reply:
When you create a non-anonymous user, that user is logged in.
If you then create an anonymous user, the client stays logged in with the previous user's credentials.